src/change/models/changeManager.ts

```diff
@@ -0,0 +1,124 @@
+import { getChangeFromPoint, getChangeFromLine, getChangeFromPolygon } from '@map-colonies/osm-change-generator';
+
+import { Services } from '../../common/constants';
+import { ILogger } from '../../common/interfaces';
+import { validateArrayHasElements } from '../../common/util';
+import { ChangeModel } from './change';
+import { OsmApiElements } from './helpers';
+import { FlattenedGeoJSON, FlattenedGeoJSONPoint, FlattenedGeoJSONLine, FlattenedGeoJSONPolygon } from './geojsonTypes';
+import { ParseOsmElementsError } from './errors';
+import { isNode, isWay } from './helpers';
+
+export const generateOsmChange = (action: Actions, feature: FlattenedGeoJSON, osmElements: OsmApiElements): OsmChange => {
+  switch (feature.geometry.type) {
+    case 'Point': {
+      feature = feature as FlattenedGeoJSONPoint;
+      switch (action) {
+        case Actions.CREATE: {
+          return getChangeFromPoint({ action, feature });
+        }
+        case Actions.MODIFY: {
+          const oldElement = getNodeFromElements(osmElements);
+          return getChangeFromPoint({ action, feature, oldElement });
+        }
+        case Actions.DELETE: {
+          const oldElement = getNodeFromElements(osmElements);
+          return getChangeFromPoint({ action, oldElement });
+        }
+      }
+    }
+    case 'LineString': {
+      feature = feature as FlattenedGeoJSONLine;
+      switch (action) {
+        case Actions.CREATE: {
+          return getChangeFromLine({ action, feature });
+        }
+        case Actions.MODIFY: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromLine({ action, feature, oldElement });
+        }
+        case Actions.DELETE: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromLine({ action, oldElement });
+        }
+      }
+    }
+    case 'Polygon': {
+      feature = feature as FlattenedGeoJSONPolygon;
+      switch (action) {
+        case Actions.CREATE: {
+          return getChangeFromPolygon({ action, feature });
+        }
+        case Actions.MODIFY: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromPolygon({ action, feature, oldElement });
+        }
+        case Actions.DELETE: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromPolygon({ action, oldElement });
+        }
+      }
+    }
+  }
+};
+
+export const getNodeFromElements = (elements: OsmApiElements): OsmNode => {
+  if (!validateArrayHasElements(elements)) {
+    throwParseOsmElementsError('node');
+  }
+  const node = elements[0];
+  if (!isNode(node)) {
+    throwParseOsmElementsError('node');
+  }
+  return node as OsmNode;
+};
+
+export const getOsmWayFromElements = (elements: OsmApiElements): OsmWay => {
+  const osmWay = parseOsmWayApi(elements);
+  if (osmWay === undefined) {
+    throwParseOsmElementsError('way');
+  }
+  return osmWay as OsmWay;
+};
```

Comment:
Try to prevent from using as like we talked about on the daily: 
```
export const getOsmWayFromElements = (elements: OsmApiElements): OsmWay => {
  const osmWay: OsmWay | undefined = parseOsmWayApi(elements);
  if (osmWay === undefined) {
    return throwParseOsmElementsError('way');
  }
  return osmWay;
};
```

Reply:
done.
it's actually a conflicting result because of the missing `return` keyword on the `throwParseOsmElementsError` call, even though it returns 'never` typescript considers the possibility of lines 83 and 85 to be both called at once.